src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -783,7 +783,7 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                 // Specify an HTTP server.
                 connectHandle = Interop.WinHttp.WinHttpConnect(
                     _sessionHandle,
-                    state.RequestMessage.RequestUri.Host,
+                    state.RequestMessage.RequestUri.HostNameType == UriHostNameType.IPv6 ? $"[{state.RequestMessage.RequestUri.IdnHost}]" : state.RequestMessage.RequestUri.IdnHost,
```

Comment:
With regard to [this](https://github.com/dotnet/corefx/pull/28818#discussion_r179288047) i'm writing a manual test.

Reply:
@davidsh @rmkerr @caesar1995 without `[]`  ipv6 test fail.

Reply:
@caesar1995 related to [PR](https://github.com/dotnet/corefx/issues/28863#issue-311719169) if this PR will be merged i think we'll need to remove workaround also here.

Reply:
Yes, thanks for bringing this up! I will update that issue with this PR.